### PR TITLE
Likes

### PR DESCRIPTION
**Pull request by DivyanshSareen**
implemented
- video page
- likes page and functionality
- history page and functionality

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -2,7 +2,10 @@ import { Routes, Route } from "react-router-dom";
 import Nav from "./components/Nav/Nav";
 import HomePage from "./routes/HomePage";
 import VideoListing from "./routes/VideoListing";
+import VideoPage from "./routes/VideoPage";
+import LikesPage from "./routes/LikesPage";
 import PlaylistManage from "./routes/PlaylistManage";
+import HistoryPage from "./routes/HistoryPage";
```
Comment (harshalini) [inline src/App.js:8]: By creating a separate file, you can import all these functions from a single file

File: src/context/Provider.js [hunk 2]
```diff
@@ -2,15 +2,21 @@ import { CategoryProvider } from "./category-context";
 import { VideoProvider } from "./video-context";
 import { PlaylistProvider } from "./playlist-context";
 import { AuthProvider } from "./auth-context";
+import { UserProvider } from "./user-context";
+import { HistoryProvider } from "./history-context";
```
Comment (harshalini) [inline src/context/Provider.js:6]: Same here

File: src/routes/VideoPage.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import axios from "axios";
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { useUser } from "../context/user-context";
+import { useHistory } from "../context/history-context";
+import RequiresAuth from "../RequiresAuth";
+import Loading from "../components/Loading/Loading";
+
+const VideoPage = () => {
+  const [video, setVideo] = useState({ _id: null, image: "loading.gif" });
```
Comment (harshalini) [inline src/routes/VideoPage.jsx:11]: Liked the idea of loading.gif as the initial state

**Review by harshalini — APPROVED** [review]
Code is clean and well structuted, naming conventions are also good

**Review by yashraj2312 — APPROVED** [review]
Clean folder structure. Looks good to merge.

File: src/App.js (hunk 1 above)
Comment (DivyanshSareen) [reply src/App.js:8]: Thank you for the detailed review, I am merging the PR for now but I will surely work on this.

**Pull request merged**